src/routes/checkout/Address/Address.js

```diff
@@ -1,5 +1,317 @@
-import React from "react";
+import React, { useState, useContext, useEffect, useRef } from "react";
 
+import { onAuthStateChanged } from "firebase/auth";
+import { AppContext } from "../../../context/app-context";
+import { UserContext } from "../../../context/user/user-context";
+import { SET_CHECKOUT_ADDRESS } from "../../../constant";
+
+import { doc, getDoc, setDoc } from "firebase/firestore";
+
+import {
+  ThemeProvider,
+  Container,
+  Typography,
+  Grid,
+  Autocomplete,
+  TextField,
+  Button,
+  CircularProgress,
+} from "@mui/material";
+import { ButtonStyles, responsiveTheme, textFieldStyles } from "./muiStyles";
 export default function Address() {
-  return <div>Address</div>;
+  const { dispatch: userDispatch, state: userState } = useContext(UserContext);
+  const { auth, db } = useContext(AppContext);
+
+  const [loading, setLoading] = useState(false);
+
+  // Reference values of inputs
+  const buildingRef = useRef();
+  const apartmentNumberRef = useRef();
+  const messageRef = useRef();
+
+  console.log(userState);
+
+  // Set shipping_address data from database in user context
+  useEffect(() => {
+    const userData = userState;
+    if (userData.shipping_address) {
+      userDispatch({
+        type: SET_CHECKOUT_ADDRESS,
+        payload: {
+          shipping_address: userData.shipping_address,
+        },
+      });
+    }
+  }, []);
+
+  // User submits new shipping address data
+  const handleSubmit = async () => {
+    setLoading(true);
+
+    let shippingData = {
+      shipping_address: {
+        campus: "UCSD",
+        building: buildingRef.current.value,
+        floor_apartment: apartmentNumberRef.current.value,
+        message: messageRef.current.value,
+      },
+    };
+
+    // Update address in context
+    userDispatch({
```

Comment:
As I mention before, the logic part, which includes the `dispatch` function, the updating database function should be in the function `useUpdateShipping` in the file `profile-context.js`. You don't want to think about how to update the Context or how to update database here. See the file `Phone.js` to see how I did it